### PR TITLE
Template Parts: Add further details to template part panel

### LINKED ISSUE #52377: Expanding template part detail panels
The detail panels for template parts are a little sparse which makes them look a bit unintentional:

Screenshot 2023-07-06 at 14 07 58

We can improve things by including available data points, for example the sync status from regular patterns, and template part area:

Template part

### PR DESCRIPTION
**Pull request by jameskoster**
Fixes #52377

## What?
Add more information to template part detail panels:

* The author for custom template parts.
* The source when template parts are added by a plugin.
* The sync status (template parts are always synced, but the user may not know this).
* Whether a template part that was added by a theme or plugin as been customised.
* The template part area.

## Why?
Template part detail panels can often be a little empty making them feel somewhat superfluous.

Displaying more information here gives the panel purpose, and helps the user better understand the template part they are viewing.

| Before | After |
| --- | --- |
| Screenshot 2023-07-10 at 16 27 24 | Screenshot 2023-07-10 at 16 32 06 |

## More examples
| Custom template part | Plugin template part |
| --- | --- |
| Screenshot 2023-07-10 at 16 33 21 | Screenshot 2023-07-10 at 16 33 32 |

## Testing Instructions
Browse to the detail panels for the following template parts and observe associated details:

### Custom template part
* Author.
* Area.
* Sync status.

### Theme template part
* Sync status.
* Area.
* Whether the template part has been customized or not.

### Plugin template part
* Sync status.
* Area.
* Source plugin.
* Whether the template part has been customized or not.

### CODE REVIEW

File: packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js
Comment (aaronrobertshaw) [inline packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js]: We could remove this `const description` and just rename the `descriptionText` variable above.

File: packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js
Comment (aaronrobertshaw) [inline packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js]: We need to be able to handle custom registered template part areas here.

File: packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js
Comment (aaronrobertshaw) [inline packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js]: The JSX fragments here aren't required as there is only a single child element.

File: packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js
Comment (aaronrobertshaw) [inline packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js]: This can be simplified to just the ternary operator, returning the appropriate span. In other words, we can drop the JSX fragment wrapper.

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for the work on these improvements @jameskoster 👍 

They test well for me overall although I did encounter one minor issue. When a custom template part area is registered and a template part is assigned to that, the area detail appears but is blank. 

Screenshot 2023-07-19 at 10 28 57 am

You can test this by:
1. Registering a custom template part area via the snippet below
2. Creating a template part via the site editor and assigning it to your custom template part area
3. Selecting that template part via the Patterns page

<details>
<summary>Snippet to register custom template part area</summary>

```php
<?php

function my_default_wp_template_part_areas( $default_area_definitions ) {
	return array_merge(
		$default_area_definitions,
		array(
			array(
				'area'        => 'my-area',
				'label'       => __( 'My Area' ),
				'icon'        => 'layout',
				'description' => 'A custom template part area',
			),
			array(
				'area'        => 'custom-area',
				'label'       => __( 'Custom Area' ),
				'icon'        => 'layout',
				'description' => 'Another custom template part area',
			),
		)
	);
}
add_filter( 'default_wp_template_part_areas', 'my_default_wp_template_part_areas' );

```
</details>

In such a case, we might need to retrieve all the template part areas from the editor store and find the one matching the custom area slug to use its label.

If you'd like, I can help update this PR later this week after I've caught up from some recent AFK?

I've also left some other inline comments for minor tweaks such as styling the "details" and "navigation" h2 elements the same, removing unnecessary JSX fragments etc.

**Comment by jameskoster** [comment]
Thank you for reviewing my hacky designer code :)

I may need your help with the custom areas. I found `useTemplatePartAreas()` but I don't know how to use it in this context.

No rush, this is not a high priority item.

**Comment by aaronrobertshaw** [comment]
> I may need your help with the custom areas. I found useTemplatePartAreas() but I don't know how to use it in this context.

Not a problem. I'll try and get this updated with my "hacky engineer's code" in between other tasks today or tomorrow 😁

File: packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js
Comment (aaronrobertshaw) [reply packages/edit-site/src/components/sidebar-navigation-screen-pattern/use-pattern-details.js]: This has been covered in https://github.com/WordPress/gutenberg/pull/52476/commits/d7a1b4efff37817093607cbaef0365b08ec61abf

**Comment by aaronrobertshaw** [comment]
I've made the tweaks to handle custom template part areas in the details. 

My only question is whether we should add a fallback label of some description in case a custom area is registered, a template part is created for it, and then the custom area is later removed. In that case, no template part area in the store would match the template part's slug, so no label would be shown in the area detail.

**Comment by jameskoster** [comment]
What actually happens in that scenario? My assumption would be that the template part becomes uncategorized, IE "General". Is that not the case? If not, perhaps a "None" or "–" fallback label as you suggested could be useful.

**Comment by jasmussen** [comment]
Looks good!

**Comment by aaronrobertshaw** [comment]
> What actually happens in that scenario? My assumption would be that the template part becomes uncategorized, IE "General". Is that not the case? 

While updating this PR I found an issue where orphaned template parts were included under the General category's count but not shown in the Patterns list. I've put up a fix for that in https://github.com/WordPress/gutenberg/pull/52961.

Combining that PR with this should allow all angles to be tested more effectively however I don't think it is necessary.

> If not, perhaps a "None" or "–" fallback label as you suggested could be useful.

The "None" or "-" options do seem better than a blank field. I wonder if there is a third option, could we use the template part's area slug instead? It might give the user more of a clue where it was assigned etc?

A comparison can be seen below:

| Slug | None |
|---|---|
| Screenshot 2023-07-26 at 2 19 16 pm | Screenshot 2023-07-26 at 3 19 50 pm |

Any preference @jameskoster or @jasmussen?

**Comment by jasmussen** [comment]
"Areas" have been confusing me so I'll defer to Jay on that.

**Comment by jameskoster** [comment]
Would it be possible to strip hyphens and sentence case the slug, and perhaps append "(Removed)"? IE:

> Area: My area (removed)

That way the user is able to interpret the original purpose of the template part, and understand that the original area no longer exists.

> "Areas" have been confusing me so I'll defer to Jay on that.

I think the term "area" makes more sense in the context of editing a template. In template part management flows "Type" might be better. Probably one to look into in a separate PR so that we capture other "area" instances like the template part creation modal.

**Comment by aaronrobertshaw** [comment]
> Would it be possible to strip hyphens and sentence case the slug, and perhaps append "(Removed)"? IE:

Certainly is. I've made this change in e6a5063afc

Screenshot 2023-07-27 at 11 40 37 am

If you are happy with the current iteration, I can approve this PR so it can be merged.

**Comment by jameskoster** [comment]
I re-ordered the navigation / details panels so that navigation appears first: 

Screenshot 2023-07-27 at 10 36 41

I'm happy if you are :) Thanks for all the help.

**Review by aaronrobertshaw — APPROVED** [review]
I've taken this for another spin and kicked off the failing test again. 

LGTM 👍

**Comment by mtias** [comment]
@aaronrobertshaw @jameskoster can we ensure we change the title of the PR before merging so that it doesn't read "Try..."? It's weird to see these in the changelog for a release. Once we are confident in the work let's reword it.

**Pull request merged**
